src/hooks/useCoauthors.ts

```diff
@@ -0,0 +1,14 @@
+import { useQuery } from '@tanstack/react-query'
+
+import { Governance } from '../clients/Governance'
+import { CoauthorStatus } from '../entities/Coauthor/types'
+
+function useCoauthors(proposalId: string, status?: CoauthorStatus) {
+  const { data: coAuthors } = useQuery({
+    queryKey: [`coauthors-${proposalId}`],
+    queryFn: () => Governance.get().getCoAuthorsByProposal(proposalId, status),
```

Comment:
You can handle here if there is no proposalId instead of passing an empty string in some cases to avoid querying it.

Reply:
https://tanstack.com/query/v3/docs/react/guides/dependent-queries